app/models/git_repository.rb

```diff
@@ -29,14 +29,16 @@ def update!(executor: TerminalExecutor.new(StringIO.new))
     executor.execute!("cd #{repo_cache_dir}", 'git fetch -p')
   end
 
-  def commit_from_ref(git_reference)
-    description = Dir.chdir(repo_cache_dir) do
-      IO.popen(['git', 'describe', '--long', '--tags', '--all', git_reference]) do |io|
+  def commit_from_ref(git_reference, length: 7)
```

Comment:
The previous behavior of the method was that it returned a SHA with a length of 7. I wanted to keep the default behavior the same.